### ISSUE TITLE
Investigate changing restecg preprocessing

### ISSUE THREAD

**Issue by jerradmgenson** [jerradmgenson/qdaim-proto#66]
Left ventricular hypertrophy shows a stronger correlation with heart disease in the Cleveland dataset than in other datasets. Therefore, we should consider updating the preprocessing for restecg so that it treats left ventricular hypertrophy differently than it treats a normal ECG.

**Comment by jerradmgenson** [comment]
I built models using QDA with robust scaling and three different combinations of restecg preprocessing: binary restecg class excluding left ventricular hypertrophy, binary restecg class including left ventricular hypertrophy (where left ventricular hypertrophy is treated the same as ST/T abnormality), and ternary restecg class. The performance of each of these three models is shown below. However, binary restecg excluding left ventricular hypertrophy performed the best on almost every metric, therefore we will keep the preprocessing as it is.

Binary restecg w/o left ventricular hypertrophy
-------------------------------------------------------------
Training dataset:      /build/training.csv
Validation dataset:    /build/validation.csv
Random number seed:    3307259
Scoring method:        informedness
Algorithm:             quadratic discriminant analysis
Preprocessing method:  robust scaling
Training samples:      458
Validation samples:    154
Generating model...
Scoring model...

Model scores:
accuracy:     0.7792
informedness: 0.5597
mcc:          0.562
precision:    0.75
recall:       0.8289
f1_score:     0.7875
ami:          0.2398
sensitivity:  0.8289
specificity:  0.7308
dor:          13.15
lr_plus:      3.079
lr_minus:     0.2341
roc_auc:      0.7799

20-fold cross-validation scores:
median accuracy:     0.8032
mad accuracy:        0.0328
median informedness: 0.6101
mad informedness:    0.06742
median mcc:          0.611
mad mcc:             0.06485
median precision:    0.8063
mad precision:       0.05208
median recall:       0.7947
mad recall:          0.05691
median f1_score:     0.8048
mad f1_score:        0.03305
median ami:          0.2739
mad ami:             0.0691
median sensitivity:  0.7947
mad sensitivity:     0.05691
median specificity:  0.8125
mad specificity:     0.04808
median dor:          18.04
mad dor:             9.958
median lr_plus:      4.304
mad lr_plus:         1.129
median lr_minus:     0.2509
mad lr_minus:        0.07509
median roc_auc:      0.805
mad roc_auc:         0.03371

Outlier scores:
accuracy:     0.6875
informedness: 0.4127
mcc:          0.4229
precision:    0.6
recall:       0.8571
f1_score:     0.7059
ami:          0.09464
sensitivity:  0.8571
specificity:  0.5556
dor:          7.5
lr_plus:      1.929
lr_minus:     0.2571
roc_auc:      0.7063
outliers:     16.0
Model hyperparameters:
{'memory': None, 'steps': [('preprocessing', RobustScaler()), ('model', QuadraticDiscriminantAnalysis())], 'verbose': False, 'preprocessing': RobustScaler(), 'model': QuadraticDiscriminantAnalysis(), 'preprocessing__copy': True, 'preprocessing__quantile_range': (25.0, 75.0), 'preprocessing__with_centering': True, 'preprocessing__with_scaling': True, 'model__priors': None, 'model__reg_param': 0.0, 'model__store_covariance': False, 'model__tol': 0.0001}


Binary restecg w/ left ventricular hypertrophy
-----------------------------------------------------------
Training dataset:      /build/training.csv
Validation dataset:    /build/validation.csv
Random number seed:    3307259
Scoring method:        informedness
Algorithm:             quadratic discriminant analysis
Preprocessing method:  robust scaling
Training samples:      458
Validation samples:    154
Generating model...
Scoring model...

Model scores:
accuracy:     0.7727
informedness: 0.5466
mcc:          0.5482
precision:    0.747
recall:       0.8158
f1_score:     0.7799
ami:          0.2267
sensitivity:  0.8158
specificity:  0.7308
dor:          12.02
lr_plus:      3.03
lr_minus:     0.2521
roc_auc:      0.7733

20-fold cross-validation scores:
median accuracy:     0.8
mad accuracy:        0.05806
median informedness: 0.6029
mad informedness:    0.111
median mcc:          0.5946
mad mcc:             0.105
median precision:    0.8125
mad precision:       0.04808
median recall:       0.7857
mad recall:          0.05944
median f1_score:     0.7991
mad f1_score:        0.04808
median ami:          0.2565
mad ami:             0.1018
median sensitivity:  0.7857
mad sensitivity:     0.05944
median specificity:  0.8125
mad specificity:     0.05515
median dor:          17.1
mad dor:             10.02
median lr_plus:      4.126
mad lr_plus:         1.237
median lr_minus:     0.2605
mad lr_minus:        0.103
median roc_auc:      0.8015
mad roc_auc:         0.05552

Outlier scores:
accuracy:     0.6875
informedness: 0.4127
mcc:          0.4229
precision:    0.6
recall:       0.8571
f1_score:     0.7059
ami:          0.09464
sensitivity:  0.8571
specificity:  0.5556
dor:          7.5
lr_plus:      1.929
lr_minus:     0.2571
roc_auc:      0.7063
outliers:     16.0
Model hyperparameters:
{'memory': None, 'steps': [('preprocessing', RobustScaler()), ('model', QuadraticDiscriminantAnalysis())], 'verbose': False, 'preprocessing': RobustScaler(), 'model': QuadraticDiscriminantAnalysis(), 'preprocessing__copy': True, 'preprocessing__quantile_range': (25.0, 75.0), 'preprocessing__with_centering': True, 'preprocessing__with_scaling': True, 'model__priors': None, 'model__reg_param': 0.0, 'model__store_covariance': False, 'model__tol': 0.0001}


Ternary restecg
---------------------
Training dataset:      /build/training.csv
Validation dataset:    /build/validation.csv
Random number seed:    3307259
Scoring method:        informedness
Algorithm:             quadratic discriminant analysis
Preprocessing method:  robust scaling
Training samples:      458
Validation samples:    154
Generating model...
Scoring model...

Model scores:
accuracy:     0.7662
informedness: 0.5331
mcc:          0.5334
precision:    0.75
recall:       0.7895
f1_score:     0.7692
ami:          0.2129
sensitivity:  0.7895
specificity:  0.7436
dor:          10.88
lr_plus:      3.079
lr_minus:     0.2831
roc_auc:      0.7665

20-fold cross-validation scores:
median accuracy:     0.8032
mad accuracy:        0.03656
median informedness: 0.6084
mad informedness:    0.07173
median mcc:          0.6001
mad mcc:             0.07841
median precision:    0.8125
mad precision:       0.05288
median recall:       0.7876
mad recall:          0.06032
median f1_score:     0.8074
mad f1_score:        0.04714
median ami:          0.2627
mad ami:             0.08679
median sensitivity:  0.7876
mad sensitivity:     0.06032
median specificity:  0.8063
mad specificity:     0.05625
median dor:          19.39
mad dor:             8.694
median lr_plus:      4.093
mad lr_plus:         1.237
median lr_minus:     0.2501
mad lr_minus:        0.08149
median roc_auc:      0.8042
mad roc_auc:         0.03586

Outlier scores:
accuracy:     0.6875
informedness: 0.4127
mcc:          0.4229
precision:    0.6
recall:       0.8571
f1_score:     0.7059
ami:          0.09464
sensitivity:  0.8571
specificity:  0.5556
dor:          7.5
lr_plus:      1.929
lr_minus:     0.2571
roc_auc:      0.7063
outliers:     16.0
Model hyperparameters:
{'memory': None, 'steps': [('preprocessing', RobustScaler()), ('model', QuadraticDiscriminantAnalysis())], 'verbose': False, 'preprocessing': RobustScaler(), 'model': QuadraticDiscriminantAnalysis(), 'preprocessing__copy': True, 'preprocessing__quantile_range': (25.0, 75.0), 'preprocessing__with_centering': True, 'preprocessing__with_scaling': True, 'model__priors': None, 'model__reg_param': 0.0, 'model__store_covariance': False, 'model__tol': 0.0001}
